### ISSUE TITLE
New settings section for addons with enable/disable/reorder option

### ISSUE THREAD

**Issue by pauloamgomes** [agentejo/cockpit#863]
Think would be interesting if we can have in the settings page a new section for addons, that just list them with some info and provide option to enable/disable/reorder (without removing them from the filesystem). 
So that way only the enable addons/modules would be loaded during the bootstrap, I can see that the loadModules() is loading everything from the addons and modules folder.
That would also permit to set dependencies (load addon x after y is initialized)

**Comment by aheinze** [comment]
I think it is a good idea to have an overview of installed addons but the other suggestions would complicate stuff. Need to think about a bigger picture here (addons installer etc).

Thanks for you suggestions

**Comment by pauloamgomes** [comment]
Yep, I understand your point, at least we can start only with the overview, simple list/grid of addons with name and info.

The other part would require more investigation, it can be similar to a market place of addons, where you can install/uninstall directly from the UI (e.g. similar to wordpress)

**Comment by raffaelj** [comment]
I like the idea. Some suggestions:

Allow deactivating core modules, too (if I don't need Singletons and Forms)

Addons could have a json file or an addon.info trigger with infos about it
  * [example of Nextcloud - markdown addon](https://github.com/icewind1991/files_markdown/blob/master/appinfo/info.xml) --> with xml files
  * could also contain info about last testet cockpit version...

@jakerb started to write a marketplace addon 10 months ago. I'm not sure, if it works and if it was exactly for installing addons, but it looks so from skimming the code.
  * https://github.com/jakerb/cockpit-module-marketplace
  * https://github.com/jakerb/cockpit-marketplace

But an actual installer seems to be a lot of work. Disabling or enabling modules through config.yaml would be enough to start with.

```yaml
modules:
    singletons  : false
addons:
    Groups      : true
    UniqueSlugs : true
    Feed        : false
```

A GUI for addon management could be an extra addon.

Some more questions, that came in my mind:

Should addons be enabled by default?
Is it better to deactivate paths or addon names?

**Comment by raffaelj** [comment]
I wrote a base to start with: https://github.com/raffaelj/cockpit/tree/cp-next_deactivate-modules-addons

I didn't change a lot of code: https://github.com/raffaelj/cockpit/commit/0191e49160b3d02dd4e7cdca7f81d5fca98aae3a

add modules to `config.yaml`

```
active.modules:
    singletons  : false
    forms       : true
    feed        : false
```

overview: `url/to/cockpit/addons/index` or via settings menu

**Comment by pauloamgomes** [comment]
Yeah, defining in the config.yaml  makes things easier and clean, in such case that will always require an update in the core bootstrap

**Comment by raffaelj** [comment]
I forgot one thought. Instead of making it really complex, the default could be to check, if an optional file exists and to include it. If not or if it returns true, the addon will be registered. Than it would be up to the user to write some custom logic.

This would be easy and simple, but it would be hard to use an addon as a GUI.

**Comment by raffaelj** [comment]
@pauloamgomes I like your approach with only three lines, but we both forgot one thing: Lime/App.php should stay as a library without any cockpit related stuff.

It's better to replace

```php
$app->loadModules(array_merge([
            COCKPIT_DIR.'/modules',  # core
            COCKPIT_DIR.'/addons' # addons
        ], $config['loadmodules'] ?? []));
```

with a function, that sets the autoload for module dirs and calls registerModule() separately per activated addon.

**Comment by pauloamgomes** [comment]
@raffaelj , yeah, you are right, but we can simplify a bit more like:

https://github.com/pauloamgomes/cockpit/commit/142515d271a02b02ac411a738652b56c195d0e2d
